ios/src/TGMapViewController.mm

```diff
@@ -1,101 +1,222 @@
 //
-//  ViewController.m
+//  TGMapViewController.mm
 //  TangramiOS
 //
 //  Created by Matt Blair on 8/25/14.
-//  Copyright (c) 2014 Mapzen. All rights reserved.
+//  Updated by Matt Smollinger on 7/29/16.
+//  Copyright (c) 2016 Mapzen. All rights reserved.
 //
 
-#import "ViewController.h"
+#import "TGMapViewController.h"
 #include "platform_ios.h"
 
-@interface ViewController () {
+@interface TGMapViewController ()
 
+@property (nullable, strong, nonatomic) EAGLContext *context;
+@property (assign, nonatomic) CGFloat pixelScale;
+@property (assign, nonatomic) BOOL renderRequested;
+
+@end
+
+
+@implementation TGMapViewController
+
+- (void)setPosition:(TangramGeoPoint)position {
+    if (self.map) {
+        self.map->setPosition(position.longitude, position.latitude);
+    }
 }
-@property (strong, nonatomic) EAGLContext *context;
-@property CGFloat pixelScale;
-@property bool renderRequested;
 
-- (void)setupGL;
-- (void)tearDownGL;
-- (void)respondToTapGesture:(UITapGestureRecognizer *)tapRecognizer;
-- (void)respondToDoubleTapGesture:(UITapGestureRecognizer *)doubleTapRecognizer;
-- (void)respondToPanGesture:(UIPanGestureRecognizer *)panRecognizer;
-- (void)respondToPinchGesture:(UIPanGestureRecognizer *)pinchRecognizer;
+- (void)setPositionEasedAtPosition:(TangramGeoPoint)position withDuration:(float)duration {
+    [self setPositionEasedAtPosition:position withDuration:duration withEaseType:TangramEaseTypeQuint];
+}
 
-@end
+- (void)setPositionEasedAtPosition:(TangramGeoPoint)position withDuration:(float)duration withEaseType:(TangramEaseType)easeType {
+    
+    if (self.map) {
+        Tangram::EaseType ease = [self convertEaseTypeFrom:easeType];
+        self.map->setPositionEased(position.longitude, position.latitude, duration, ease);
+    }
+}
+
+- (TangramGeoPoint)position {
+    TangramGeoPoint returnVal;
+    if (self.map){
+        self.map->getPosition(returnVal.longitude, returnVal.latitude);
+        return returnVal;
+    }
+    //Null Island
+    returnVal.latitude = 0.0;
+    returnVal.longitude = 0.0;
+    return returnVal;
+}
+
+- (void)setZoom:(float)zoom {
+    if (self.map) {
+        self.map->setZoom(zoom);
+    }
+}
+
+- (Tangram::EaseType)convertEaseTypeFrom:(TangramEaseType)ease {
+    switch (ease) {
+        case TangramEaseTypeLinear:
+            return Tangram::EaseType::linear;
+        case TangramEaseTypeSine:
+            return Tangram::EaseType::sine;
+        case TangramEaseTypeQuint:
+            return Tangram::EaseType::quint;
+        case TangramEaseTypeCubic:
+            return Tangram::EaseType::cubic;
+        default:
+            return Tangram::EaseType::quint;
+    }
+}
+
+- (void)setZoomEased:(float)zoomLevel withDuration:(float)duration {
+    [self setZoomEased:zoomLevel withDuration:duration withEaseType:TangramEaseTypeQuint];
```

Comment:
Default for ease is `Cubic` on android: https://github.com/tangrams/tangram-es/blob/master/android/tangram/src/com/mapzen/tangram/MapController.java#L51